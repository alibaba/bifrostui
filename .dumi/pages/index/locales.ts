export default {
  zhCN: {
    description: '一款简洁优雅的跨端组件库',
    start: '开始体验',
    features: [
      {
        title: '一码多端',
        desc: '同时支持H5和小程序',
      },
      {
        title: '高定制',
        desc: '一致优雅的API帮助您快速创造新主题',
      },
      {
        title: '高可用',
        desc: '完善的配套，丰富的组件，开箱即用',
      },
    ],
    subtitle: [
      {
        title: '设计语言',
        desc: '基于阿里影业的众多业务实践沉淀的设计语言和设计系统',
        btn: '查看详情',
      },
      {
        title: '组件列表',
        desc: '一致的API、丰富的组件帮助您快速构建前端应用',
        btn: '查看详情',
      },
      {
        title: '谁在使用',
      },
    ],
  },
  enUS: {
    description: 'A concise and flexible cross-end component library',
    start: 'Start to experience',
    features: [
      {
        title: 'One code for multiple terminals',
        desc: 'Supports both H5 and mini programs',
      },
      {
        title: 'High customization',
        desc: 'Consistent and elegant API helps you quickly create new themes',
      },
      {
        title: 'High availability',
        desc: 'Complete supporting facilities, rich components, ready to use',
      },
    ],
    subtitle: [
      {
        title: 'Design language',
        desc: 'Design language and design system based on the numerous business practices of Alibaba Pictures',
        btn: 'View details',
      },
      {
        title: 'Component list',
        desc: 'Consistent API and rich components help you quickly build front-end applications',
        btn: 'View details',
      },
      {
        title: 'Who is using it',
      },
    ],
  },
};
