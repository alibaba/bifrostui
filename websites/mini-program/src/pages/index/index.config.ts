export default definePageConfig({
  navigationBarTitleText: 'BUI小程序',
});
