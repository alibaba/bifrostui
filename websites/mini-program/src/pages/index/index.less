.home-wrapper {
  padding: 6px 12px 30px;

  .header {
    display: flex;
    align-items: center;
    flex-direction: column;
    margin-bottom: 12px;

    .logo {
      width: 104px;
      height: 30px;
      margin-right: 12px;
    }

    .desc {
      .desc-title {
        font-size: 16px;
        font-weight: 500;
        margin-bottom: 8px;
      }

      .desc-summary {
        font-size: 14px;
        color: #999;
        margin-top: 10px;
      }
    }
  }

  .group-wrapper {
    .group-item {
      margin-bottom: 12px;

      .group-name {
        font-size: 16px;
        font-weight: 500;
        padding: 9px 4px;
        background-color: var(--bui-color-bg-default);
        margin-bottom: 8px;
      }

      .component-list {
        padding: 0 4px;
        color: #484f58;

        .component-item {
          padding: 9px;
          margin-bottom: 8px;
          border: 1px solid rgba(233, 235, 239, 0.5);
          border-radius: 4px;
        }
      }
    }
  }
}
