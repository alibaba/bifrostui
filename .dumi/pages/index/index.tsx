/* eslint-disable react/display-name */
import React from 'react';
import { Button } from '@bifrostui/react';
import { usePrefersColor, useNavigate, useIntl, useLocale } from 'dumi';
import { ToTopOutlinedIcon } from '@bifrostui/icons';
import Tpp from './user-icon/tpp';
import Dm from './user-icon/dm';
import './index.less';
import locales from './locales';

export default () => {
  const [color] = usePrefersColor();
  const { locale } = useIntl();
  const current = locale === 'zh-CN' ? 'zhCN' : 'enUS';
  const navigate = useNavigate();
  return (
    <div className="main-container home-page">
      <div className="main-head">
        <div>
          <img
            className="main-logo"
            src="https://gw.alicdn.com/imgextra/i2/O1CN01Bqi4MA1USNSLdER5C_!!6000000002516-55-tps-190-90.svg"
          />
          <div className="main-desc">{locales[current].description}</div>
          <Button
            size="large"
            variant="contained"
            className="start-btn"
            endIcon={<ToTopOutlinedIcon />}
            onClick={() => {
              navigate('/guide/introduce', { replace: true });
            }}
          >
            {locales[current].start}
          </Button>
        </div>
        <img
          className="head-img"
          src="https://gw.alicdn.com/imgextra/i3/O1CN018hn1121UWUwJyUdB0_!!6000000002525-2-tps-960-960.png"
        />
      </div>
      <div className="main-introduce">
        <div className="introduce-item">
          <img
            className="introduce-icon"
            src="https://gw.alicdn.com/imgextra/i1/O1CN01y4iaZr1aLP2g7R9xm_!!6000000003313-2-tps-160-160.png"
          />
          <div className="introduce-title">
            {locales[current].features[0].title}
          </div>
          <div className="introduce-desc">
            {locales[current].features[0].desc}
          </div>
        </div>
        <div className="introduce-item">
          <img
            className="introduce-icon"
            src="https://gw.alicdn.com/imgextra/i4/O1CN01fvzf101QVLTsWkcrR_!!6000000001981-2-tps-160-160.png"
          />
          <div className="introduce-title">
            {locales[current].features[1].title}
          </div>
          <div className="introduce-desc">
            {locales[current].features[1].desc}
          </div>
        </div>
        <div className="introduce-item">
          <img
            className="introduce-icon"
            src="https://gw.alicdn.com/imgextra/i4/O1CN01CCqAR028qGfPoYba0_!!6000000007983-2-tps-160-160.png"
          />
          <div className="introduce-title">
            {locales[current].features[2].title}
          </div>
          <div className="introduce-desc">
            {locales[current].features[2].desc}
          </div>
        </div>
      </div>
      <div className="main-resource">
        <div className="resource-item">
          <div className="resource-title">
            {locales[current].subTitle[0].title}
          </div>
          <div className="resource-content">
            <div>
              <div className="content-title">
                {locales[current].subTitle[0].title}
              </div>
              <div className="content-desc">
                {locales[current].subTitle[0].desc}
              </div>
              <Button
                size="large"
                variant="contained"
                style={{ '--bg-color': '#2356df', '--text-color': '#fff' }}
                endIcon={<ToTopOutlinedIcon />}
                onClick={() => {
                  navigate('/design/colors', { replace: true });
                }}
              >
                {locales[current].subTitle[0].btn}
              </Button>
            </div>
            <img
              className="content-img"
              src="https://gw.alicdn.com/imgextra/i3/O1CN016LMfu721Ngdq3YrzQ_!!6000000006973-2-tps-600-600.png"
            />
          </div>
        </div>
        <div className="resource-item">
          <div className="resource-title">
            {locales[current].subTitle[1].title}
          </div>
          <div className="resource-content">
            <div>
              <div className="content-title">
                {locales[current].subTitle[1].title}
              </div>
              <div className="content-desc">
                {locales[current].subTitle[1].desc}
              </div>
              <Button
                size="large"
                variant="contained"
                style={{ '--bg-color': '#2356df', '--text-color': '#fff' }}
                endIcon={<ToTopOutlinedIcon />}
                onClick={() => {
                  navigate('/cores/button', { replace: true });
                }}
              >
                {locales[current].subTitle[1].btn}
              </Button>
            </div>
            <img
              className="content-img"
              src="https://gw.alicdn.com/imgextra/i3/O1CN01DJfWf71xAcssFhdD6_!!6000000006403-2-tps-600-600.png"
            />
          </div>
        </div>
      </div>

      <div className="main-users">
        <div className="users-title">{locales[current].subTitle[2].title}</div>
        <div className="user-list">
          <Tpp color={color === 'light' ? '#eee' : '#5F6672'} />
          <Dm color={color === 'light' ? '#eee' : '#5F6672'} />
        </div>
      </div>
      <div className="main-footer">
        Open-source MIT Licensed | Copyright Â© 2023-present Powered by BUI
      </div>
    </div>
  );
};
