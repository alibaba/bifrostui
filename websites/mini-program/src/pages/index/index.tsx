import menuGroups from '@/route.config';
import Taro, { useShareAppMessage, useShareTimeline } from '@tarojs/taro';
import './index.less';

const Index = () => {
  useShareAppMessage(() => {
    return {
      title: 'BUI小程序',
      path: 'pages/index/index',
    };
  });

  useShareTimeline(() => {
    return {
      title: 'BUI小程序',
      path: 'pages/index/index',
    };
  });

  const goDemo = (component) => {
    Taro.navigateTo({
      url: `/${component.path}`,
    });
  };

  return (
    <div className="home-wrapper ">
      <div className="header">
        <img
          className="logo"
          src="https://gw.alicdn.com/imgextra/i4/O1CN01XWp8e31QbIi5YgLUw_!!6000000001994-2-tps-362-96.png"
        />
        <div className="desc">
          <span className="desc-summary">
            BUI React-阿里影业风格的跨端组件库
          </span>
        </div>
      </div>
      <div className="group-wrapper">
        {menuGroups.map((group) => (
          <div className="group-item" key={group.groupEnName}>
            <div className="group-name">{group.groupName}</div>
            <div className="component-list">
              {(group.components || [])
                .sort((a, b) => b.order - a.order)
                .map((component) => (
                  <div
                    className="component-item"
                    key={component.enName}
                    onClick={() => goDemo(component)}
                  >
                    {component.zhName}({component.enName})
                  </div>
                ))}
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Index;
