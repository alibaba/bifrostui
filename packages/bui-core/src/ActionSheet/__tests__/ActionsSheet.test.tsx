import React from 'react';
import { act, isConformant, render, screen, userEvent } from 'testing';
import { ActionSheet, ActionSheetItem } from '../index';

describe('ActionSheet', () => {
  const rootClass = 'bui-action-sheet';
  beforeEach(() => {
    jest.useFakeTimers();
  });
  afterEach(() => {
    jest.clearAllTimers();
    jest.useRealTimers();
  });

  isConformant({
    Component: ActionSheet,
    displayName: 'BuiActionSheet',
    className: rootClass,
    requiredProps: {
      open: true,
    },
    getTargetElement: () => {
      return document.querySelector('.bui-action-sheet');
    },
  });
  it('renders correctly', async () => {
    const { container } = render(
      <ActionSheet open disablePortal extra="extra action item">
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem>action item edit</ActionSheetItem>
        <ActionSheetItem disabled>action item disabled</ActionSheetItem>
        <ActionSheetItem bold>action item bold</ActionSheetItem>
        <ActionSheetItem description="description">
          action item description
        </ActionSheetItem>
      </ActionSheet>,
    );
    await act(async () => {
      await jest.runAllTimers();
    });
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should open and close', async () => {
    const { rerender } = render(
      <ActionSheet>
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem>action item edit</ActionSheetItem>
        <ActionSheetItem>action item delete</ActionSheetItem>
      </ActionSheet>,
    );
    expect(screen.queryAllByText(/action item/).length).toBe(0);
    rerender(
      <ActionSheet open>
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem>action item edit</ActionSheetItem>
        <ActionSheetItem>action item delete</ActionSheetItem>
      </ActionSheet>,
    );
    expect(screen.queryAllByText(/action item/).length).toBe(3);
  });
  it('should hide backdrop with hideBackdrop props', () => {
    render(
      <ActionSheet open hideBackdrop>
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem>action item edit</ActionSheetItem>
      </ActionSheet>,
    );
    expect(document.querySelector('.bui-backdrop')).not.toBeInTheDocument();
  });
  it('render extra element in first child', () => {
    render(
      <ActionSheet open extra="extra action item">
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem>action item edit</ActionSheetItem>
      </ActionSheet>,
    );
    const index = screen.getAllByText(/action item/).findIndex((item) => {
      return item.textContent === 'extra action item';
    });
    expect(index).toBe(0);
  });

  it('onClick can not be called when actions item disabled', () => {
    const onClick = jest.fn();
    render(
      <ActionSheet open>
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem disabled onClick={onClick} data-testid="disable-item">
          action item edit
        </ActionSheetItem>
      </ActionSheet>,
    );
    const disableItem = screen.getByTestId('disable-item');
    expect(disableItem).toHaveClass(`${rootClass}-item-disabled`);
    userEvent.click(disableItem);
    expect(onClick).not.toHaveBeenCalled();
  });

  it('actions item onclick function should be called', () => {
    const onClick = jest.fn();
    render(
      <ActionSheet open>
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem onClick={onClick}>action item edit</ActionSheetItem>
      </ActionSheet>,
    );
    userEvent.click(screen.getByText(/action item edit/));
    expect(onClick).toHaveBeenCalled();
  });

  it('actions item with bold can render correctly', () => {
    render(
      <ActionSheet open>
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem bold data-testid="bold-item">
          action item edit
        </ActionSheetItem>
      </ActionSheet>,
    );
    expect(screen.getByTestId('bold-item')).toHaveClass(
      `${rootClass}-item-bold`,
    );
  });
  it('actions item with description can render correctly', () => {
    render(
      <ActionSheet open>
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem description="description">
          action item edit
        </ActionSheetItem>
      </ActionSheet>,
    );
    expect(screen.getByText(/description/)).toHaveClass(
      `${rootClass}-item-description`,
    );
  });
  it.each([
    'primary',
    'warning',
    'success',
    'info',
    'danger',
    'default',
  ] as const)(
    'action item should render correctly with color props',
    (color) => {
      render(
        <ActionSheet open>
          <ActionSheetItem color={color} data-testid="color-item">
            action item edit
          </ActionSheetItem>
        </ActionSheet>,
      );
      expect(screen.getByTestId('color-item')).toHaveClass(
        `${rootClass}-item-${color}`,
      );
    },
  );
  it('action item triggers click event', () => {
    const onClick = jest.fn((_, { index }) => index);
    render(
      <ActionSheet open>
        <ActionSheetItem index={2} onClick={onClick}>
          action item edit
        </ActionSheetItem>
      </ActionSheet>,
    );
    userEvent.click(screen.getByText('action item edit'));
    expect(onClick).toHaveBeenCalled();
    expect(onClick).toReturnWith(2);
  });
  it('When action item triggers a click event onSelect will respond', () => {
    const onSelect = jest.fn((_, { index }) => index);
    render(
      <ActionSheet onSelect={onSelect} open>
        <ActionSheetItem index={1}>action item copy</ActionSheetItem>
        <ActionSheetItem index={2}>action item edit</ActionSheetItem>
      </ActionSheet>,
    );
    userEvent.click(screen.getByText('action item edit'));
    expect(onSelect).toHaveBeenCalled();
    expect(onSelect).toReturnWith(2);
  });

  it('last render cancelText', () => {
    render(
      <ActionSheet open cancelText="action item cancel">
        <ActionSheetItem>action item copy</ActionSheetItem>
        <ActionSheetItem bold>action item edit</ActionSheetItem>
      </ActionSheet>,
    );
    const actionItems = screen.getAllByText(/action item/);
    const index = actionItems.findIndex((item) => {
      return item.textContent === 'action item cancel';
    });
    expect(index).toBe(actionItems.length - 1);
    expect(document.querySelector(`.${rootClass}-cancel`)).toBeInTheDocument();
  });

  describe('handle onClose function', () => {
    it('onClose callback not work with default', () => {
      const onClose = jest.fn();
      render(
        <ActionSheet onClose={onClose} open>
          <ActionSheetItem>action item copy</ActionSheetItem>
          <ActionSheetItem>action item edit</ActionSheetItem>
        </ActionSheet>,
      );
      userEvent.click(screen.getByText(/action item edit/));
      expect(onClose).not.toHaveBeenCalled();
    });
    it('onClose should be called width closeOnAction equal true', () => {
      const onClose = jest.fn();
      render(
        <ActionSheet onClose={onClose} open cancelText="cancel">
          <ActionSheetItem>action item copy</ActionSheetItem>
          <ActionSheetItem>action item edit</ActionSheetItem>
        </ActionSheet>,
      );
      userEvent.click(screen.getByText('cancel'));
      expect(onClose).toHaveBeenCalled();
    });
  });
});
