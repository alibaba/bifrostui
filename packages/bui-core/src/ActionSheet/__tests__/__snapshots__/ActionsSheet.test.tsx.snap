// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet renders correctly 1`] = `
<div
  class="bui-modal bui-drawer bui-action-sheet"
>
  <div
    class="bui-backdrop bui-modal-backdrop"
    style="visibility: visible; transition: opacity 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; opacity: 1;"
  />
  <div
    class="bui-drawer-content bui-drawer-content-bottom bui-action-sheet-content"
    style="visibility: visible; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transform: none;"
  >
    <div
      class="bui-action-sheet-extra"
    >
      extra action item
    </div>
    <div
      class="bui-action-sheet-item-wrapper"
    >
      <div
        class="bui-action-sheet-item bui-action-sheet-item-default"
      >
        <div
          class="bui-action-sheet-item-content"
        >
          action item copy
        </div>
      </div>
      <div
        class="bui-action-sheet-item bui-action-sheet-item-default"
      >
        <div
          class="bui-action-sheet-item-content"
        >
          action item edit
        </div>
      </div>
      <div
        class="bui-action-sheet-item bui-action-sheet-item-default bui-action-sheet-item-disabled"
      >
        <div
          class="bui-action-sheet-item-content"
        >
          action item disabled
        </div>
      </div>
      <div
        class="bui-action-sheet-item bui-action-sheet-item-default bui-action-sheet-item-bold"
      >
        <div
          class="bui-action-sheet-item-content"
        >
          action item bold
        </div>
      </div>
      <div
        class="bui-action-sheet-item bui-action-sheet-item-default"
      >
        <div
          class="bui-action-sheet-item-content"
        >
          action item description
        </div>
        <div
          class="bui-action-sheet-item-description"
        >
          description
        </div>
      </div>
    </div>
  </div>
</div>
`;
